app/org/sagebionetworks/bridge/services/AuthenticationService.java

```diff
@@ -116,13 +127,21 @@ public void requestEmailSignIn(SignIn signIn) {
             throw new LimitExceededException("Email currently pending confirmation.");
         }
         
-        // check that email is in the study, if not, return quietly to prevent session enumeration attacks
+        // check that email is in the study, if not, return quietly to prevent account enumeration attacks
         if (accountDao.getAccountWithEmail(study, signIn.getEmail()) == null) {
+            try {
+                // The not found case returns *much* faster than the normal case. To prevent account enumeration 
+                // attacks, measure time of a successful case and delay for that period before returning.
+                LOG.info("Email submitted for sign in that is not registered: " + signIn.getEmail());
```

Comment:
I'm not sure if we should be logging this. On the one hand, this only logs email addresses that we know are _not_ part of the study. On the other hand, we probably shouldn't be logging email addresses into the logs in general.